src/scippneutron/io/sqw/_models.py

```diff
@@ -0,0 +1,457 @@
+            "u": ir.Array(self.u.values, ty=ir.TypeTag.f64),
+            "v": ir.Array(self.v.values, ty=ir.TypeTag.f64),
+            "omega": ir.F64(_angle_value(self.omega)),
+            "dpsi": ir.F64(_angle_value(self.dpsi)),
+            "gl": ir.F64(_angle_value(self.gl)),
+            "gs": ir.F64(_angle_value(self.gs)),
+            "angular_is_degree": ir.Logical(False),
+            # serial_name and version are serialized by SqwMultiIXExperiment
+        }
+
+
+@dataclass(slots=True)
+class SqwMultiIXExperiment(ir.Serializable):
+    array_dat: list[SqwIXExperiment]
+
+    serial_name: ClassVar[str] = "IX_experiment"
+    version: ClassVar[float] = 3.0
+
+    def _serialize_to_dict(
+        self,
+    ) -> dict[str, ir.Object | ir.ObjectArray | ir.CellArray]:
+        return {
+            "serial_name": ir.String(self.serial_name),
+            "version": ir.F64(self.version),
+            "array_dat": ir.ObjectArray(
+                ty=ir.TypeTag.struct,
+                shape=(len(self.array_dat),),
+                data=[exp.serialize_to_ir() for exp in self.array_dat],
+            ),
+        }
+
+
+@dataclass(kw_only=True, slots=True)
+class UniqueRefContainer(ir.Serializable):
+    global_name: str
+    objects: UniqueObjContainer
+
+    serial_name: ClassVar[str] = "unique_references_container"
+    version: ClassVar[float] = 1.0
+
+    def _serialize_to_dict(
+        self,
+    ) -> dict[str, ir.Object | ir.ObjectArray | ir.CellArray]:
+        return {
+            "serial_name": ir.String(self.serial_name),
+            "version": ir.F64(self.version),
+            "stored_baseclass": ir.String(self.objects.baseclass),
+            "global_name": ir.String(self.global_name),
+            "unique_objects": self.objects.serialize_to_ir().to_object_array(),
+        }
+
+
+@dataclass(kw_only=True, slots=True)
+class UniqueObjContainer(ir.Serializable):
+    baseclass: str
+    objects: list[ir.Serializable]
+    indices: list[int]
+
+    serial_name: ClassVar[str] = "unique_objects_container"
+    version: ClassVar[float] = 1.0
+
+    def _serialize_to_dict(
+        self,
+    ) -> dict[str, ir.Object | ir.ObjectArray | ir.CellArray]:
+        return {
+            "serial_name": ir.String(self.serial_name),
+            "version": ir.F64(self.version),
+            "baseclass": ir.String(self.baseclass),
+            "unique_objects": ir.CellArray(
+                shape=(len(self.objects),),
+                data=[obj.serialize_to_ir().to_object_array() for obj in self.objects],
+            ),
+            "idx": ir.ObjectArray(
+                shape=(len(self.indices),),
+                # +1 to convert to 1-based indexing
+                data=np.array(self.indices) + 1.0,
+                ty=ir.TypeTag.f64,
+            ),
+        }
+
+
+def _angle_value(x: sc.Variable) -> float:
+    return x.to(unit="rad", dtype="float64", copy=False).value  # type: ignore[no-any-return]
```

Comment:
```suggestion
    return float(x.to(unit="rad", dtype="float64", copy=False))
```
might avoid the mypy bypass? Or does that raise if not dimensionless? So maybe divide by `sc.scalar(1, unit='rad')`? Or leave it, not too important.